src/ring_theory/integral_domain.lean

```diff
@@ -133,20 +133,16 @@ end
 
 variables [fintype G]
 
-lemma card_fiber_eq_of_mem_range {H : Type*} [group H] [decidable_eq H]
-  (f : G →* H) {x y : H} (hx : x ∈ set.range f) (hy : y ∈ set.range f) :
+@[to_additive] lemma monoid_hom.card_fiber_eq_of_mem_range {M} [monoid M] [decidable_eq M]
+  (f : G →* M) {x y : M} (hx : x ∈ set.range f) (hy : y ∈ set.range f) :
   (univ.filter $ λ g, f g = x).card = (univ.filter $ λ g, f g = y).card :=
```

Comment:
We should probably make an equivalence `f⁻¹ {x} ≃`[`f.ker`](https://leanprover-community.github.io/mathlib_docs/group_theory/subgroup/basic.html#monoid_hom.ker) given `x ∈ set.range f`.